app/controllers/adopted_controller.rb

```diff
@@ -0,0 +1,46 @@
+class AdoptedController < ApplicationController
+  before_action :authenticate
+  before_action :get_adopted_things, :make_cur_page, :make_other_pages, only: [:index]
```

Comment:
Given these filters are only used for the `index` action, what do you think of just moving the logic in there? You could still leave the helper functions and call them.